// mathball/lib/index.js
'use strict';

/* Number Checkers */
var armstrong = require('./armstrong'),
    automorphic = require('./automorphic'),
    kaprekar = require('./kaprekar'),
    lucky = require('./lucky'),
    neon = require('./neon'),
    palindrome = require('./palindrome'),
    prime = require('./prime'),
    smith = require('./smith');

/* Find nth Number */
var catalan = require('./catalan'),
    factorial = require('./factorial'),
    fibonacci = require('./fibonacci'),
    magic = require('./magic'),
    smart = require('./smart');

/* Mathematical Utilities */
var gcd = require('./gcd'),
    isEven = require('./isEven'),
    isOdd = require('./isOdd'),
    lcm = require('./lcm'),
    length = require('./length'),
    max = require('./max'),
    min = require('./min'),
    sort = require('./sort'),
    sum = require('./sum');

/* Helper Methods */
var performance = require('./performance'),
    pipe = require('./pipe');

/* M - Mathball Object */
module.exports = {
	isArmstrong: armstrong.isArmstrong,
	isAutomorphic: automorphic.isAutomorphic,
	catalan: catalan.catalan,
	factorial: factorial.factorial,
	fibonacci: fibonacci.fibonacci,
	gcd: gcd,
	isEven: isEven,
	isOdd: isOdd,
	isKaprekar: kaprekar.isKaprekar,
	lcm: lcm,
	length: length,
	isLucky: lucky.isLucky,
	magic: magic.magic,
	max: max,
	min: min,
	isNeon: neon.isNeon,
	isPalindrome: palindrome.isPalindrome,
	performance: performance,
	pipe: pipe,
	isPrime: prime.isPrime,
	smart: smart.smart,
	isSmith: smith.isSmith,
	sort: sort,
	sum: sum
};
